accelerometer/summariseEpoch.py

```diff
@@ -406,8 +417,123 @@ def calculateECDF(e, inputCol, summary):
         summary[inputCol + '-ecdf-' + str(accUtils.formatNum(x,0)) + 'mg'] = \
             accUtils.formatNum(ecdf, 5)
 
+def calculatePSD(e, epochPeriod, fourierWithAcc, labels, summary):
+    """Calculate the power spectral density from fourier analysis of a 1 day frequency
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param int epochPeriod: Size of epoch time window (in seconds)
+    :param bool fourierWithAcc:True calculates fourier done with acceleration data instead of sleep data
+    :param list(str) labels: Activity state labels
+    :param dict summary: Output dictionary containing all summary metrics
+
+    :return: Write dict <summary> keys 'PSD-<W/Hz>'
+    """
+    if fourierWithAcc:
+        y = (e['accImputed'])
+    else:
+        cols = []
+        # get imputed variable names for each activity type
+        # NB - make sure sleep comes first
+        for accType in labels:
+            col = accType + 'Imputed'
+            cols += [col]
+      
+        idx = cols.index('sleepImputed') #gets index of sleep label
+        # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+        y = (np.argmax(e[cols].values, axis=1) == idx).astype('int')*2-1
+        
+    n = len(y)
+    k = len(y)*epochPeriod/(60*60*24)
+    e = -2.j * np.pi * k * np.arange(n) / n
+    # finds the power spectral density for a one day cycle using fourier analysis 
+    res = np.sum(np.exp(e) * y, axis=-1)/n
+    PSD = np.abs(res)**2
+    summary['PSD'] = PSD
+
+def calculateFourierFreq(e, epochPeriod, fourierWithAcc, labels, summary):
+    """Calculate the most prevalent frequency in a fourier analysis 
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param int epochPeriod: Size of epoch time window (in seconds)
+    :paran bool fourierWithAcc: True calculates fourier done with acceleration data instead of sleep data
+    :param list(str) labels: Activity state labels
+    :param dict summary: Output dictionary containing all summary metrics
 
+    :return: Write dict <summary> keys 'fourier frequency-<1/days>'
+    """
+    if fourierWithAcc:
+        y = (e['accImputed'])
+    else:
+        cols = []
+        # get imputed variable names for each activity type
+        # NB - make sure sleep comes first
+        for accType in labels:
+            col = accType + 'Imputed'
+            cols += [col]
+            
+        idx = cols.index('sleepImputed') #gets index of sleep label
+        # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+        y = (np.argmax(e[cols].values, axis=1) == idx).astype('int')*2-1
```

Comment:
See comment in same section of code for PSD block

Reply:
see comment there